src/test/java/org/servantscode/commons/search/SearchTest.java

```diff
@@ -0,0 +1,223 @@
+package org.servantscode.commons.search;
+
+import org.junit.Test;
+
+import java.sql.Timestamp;
+import java.time.*;
+import java.time.format.DateTimeFormatter;
+import java.util.Iterator;
+
+import static junit.framework.TestCase.assertEquals;
+
+public class SearchTest {
+
+    public SearchTest() {
+    }
+
+    @Test
+    public void testSearchTextClause() {
+        Search search = new Search();
+        search.addClause(new Search.TextClause("field", "value"));
+        Search.SearchClause clause = search.getClauses().get(0);
+        assertEquals("Wrong Query.", "field ILIKE '%value%'", clause.getQuery());
+        assertEquals("Wrong SQL.", "field ILIKE ?", clause.getSql());
+        assertEquals("Wrong Values.", "[%value%]", clause.getValues().toString());
+
+        clause = new Search.TextClause("Robert◙⌐⌐x╘'\");", "\\\\\\\\\"); \\\"");
+        assertEquals("Wrong Query", "Robert◙⌐⌐x╘'\"); ILIKE '%\\\\\\\\\"); \\\"%'", clause.getQuery());
+        assertEquals("Wrong SQL.", "Robert◙⌐⌐x╘'\"); ILIKE ?", clause.getSql());
+        assertEquals("Wrong Values", "[%\\\\\\\\\"); \\\"%]", clause.getValues().toString());
+
+        clause = new Search.TextClause(null, "");
+        assertEquals("Wrong Query", "null ILIKE '%%'", clause.getQuery());
+        assertEquals("Wrong SQL.", "null ILIKE ?", clause.getSql());
+        assertEquals("Wrong Values", "[%%]", clause.getValues().toString());
+    }
+
+    @Test
+    public void testSearchIntegerClause() {
+        Search.SearchClause clause = new Search.IntegerClause("field", Integer.MAX_VALUE);
+        assertEquals("Wrong Query.", "field = 2147483647", clause.getQuery());
+        assertEquals("Wrong SQL.", "field = ?", clause.getSql());
+        assertEquals("Wrong Values", "[2147483647]", clause.getValues().toString());
+
+        clause = new Search.IntegerClause("field", Integer.MIN_VALUE);
+        assertEquals("Wrong Query.", "field = -2147483648", clause.getQuery());
+        assertEquals("Wrong SQL.", "field = ?", clause.getSql());
+        assertEquals("Wrong Values", "[-2147483648]", clause.getValues().toString());
+
+        for (int i = -100000; i < 100000; i++) {

```

Comment:
Boom! Thorough but over kill. Hit a negative number, 0 and a positive number + the MIN and MAX.